zilliqa/src/exec.rs

```diff
@@ -356,11 +356,19 @@ impl DatabaseRef for &State {
     }
 
     fn block_hash_ref(&self, number: u64) -> Result<B256, Self::Error> {
-        Ok(self
-            .block_store
-            .get_block_by_number(number)?
-            .map(|block| B256::new(block.hash().0))
-            .unwrap_or_default())
+        if let Some(current_block_number) = self.block_store.get_highest_block_number()? {
```

Comment:
We shouldn't use `get_highest_block_number` here. We might be executing a transaction on a much older block for (e.g.) replay.

Instead, we should use the block number passed into the `TxEnv`. I believe this is what `revm` does by default (see `fn block_hash` in `context.rs`), so maybe just trust it and remove the check entirely?